### PR TITLE
Re-org DVM files structure

### PR DESCRIPTION
**Pull request by AsceticBear**
No logic changes. 

## Companion Note

- Rename `dvm-ethereum` to `darwinia-ethereum`
- Rename `dc-rpc` to `darwinia-client-rpc`

### CODE REVIEW

**Comment by AsceticBear** [comment]
@HackFisher @AurevoirXavier Can we merge this today? This pr review is much simple.

File: client/rpc/Cargo.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ description = "Ethereum RPC (web3) compatibility layer for Darwinia."
 edition     = "2021"
 homepage    = "https://darwinia.network"
 license     = "GPL-3.0"
-name        = "dc-rpc"
+name        = "darwinia-rpc"
```
Comment (AurevoirXavier) [inline client/rpc/Cargo.toml:7]: This name is not good.

https://github.com/darwinia-network/darwinia/blob/main/rpc/Cargo.toml#L7

File: client/rpc-core/Cargo.toml [hunk 2]
```diff
@@ -4,7 +4,7 @@ description = "Runtime primitives for Ethereum RPC (web3) compatibility layer fo
 edition     = "2021"
 homepage    = "https://darwinia.network"
 license     = "GPL-3.0"
-name        = "dvm-rpc-core"
+name        = "darwinia-rpc-core"
```
Comment (AurevoirXavier) [inline client/rpc-core/Cargo.toml:7]: Also.

File: client/rpc/Cargo.toml (hunk 1 above)
Comment (AsceticBear) [reply client/rpc/Cargo.toml:7]: Make sense, I will update this later.

Comment (AurevoirXavier) [reply client/rpc/Cargo.toml:7]: Something like `darwinia-client-rpc`

Comment (AsceticBear) [reply client/rpc/Cargo.toml]: @AurevoirXavier Take a review again.

**Review by AurevoirXavier — APPROVED** [review]
LGTM.
Better to merge #1129 first. Then I can make some updates easily.

**Review by AurevoirXavier — APPROVED** [review]

**Pull request merged**